package/libpiksi/libpiksi/src/util.c

```diff
@@ -99,6 +113,123 @@ bool device_is_duro(void)
                  strlen(DEVICE_DURO_ID_STRING)) == 0);
 }
 
+static bool device_has_ins(void)
+{
+  return access(SMOOTHPOSE_LICENSE_FILE_PATH, F_OK) != -1;
+}
+
+int hw_version_string_get(char *hw_version_string, size_t size)
+{
+  char raw_hw_ver_string[DEVICE_HARDWARE_VERSION_MAX_LENGTH];
+  u64 hw_version = 0;
+  if (file_read_string(DEVICE_HARDWARE_VERSION_FILE_PATH, raw_hw_ver_string,
+                       sizeof(raw_hw_ver_string)) == 0) {
+    hw_version = (u64)strtod(raw_hw_ver_string, NULL);
+  } else {
+    piksi_log(LOG_ERR, "Error reading hardware version string (buffer size: %d)", sizeof(raw_hw_ver_string));
+    return -1;
+  }
+  u16 major_ver = hw_version >> 16;
+  u16 minor_ver = hw_version & 0xFFFF;
+  int written = snprintf(hw_version_string, size, "%d.%d", major_ver, minor_ver);
+  if (written < 0) {
+    piksi_log(LOG_ERR, "Error writing hardware version string to buffer");
+    return -1;
+  }
+  if (written >= size) {
+    piksi_log(LOG_ERR, "Hardware version string truncated when writing to buffer (size of intended string: %d)", written);
+    return -1;
+  }
+  return 0;
+}
+
+int hw_revision_string_get(char *hw_revision_string, size_t size)
+{
+  const char *s = NULL;
+
+  char raw_hw_rev_string[DEVICE_HARDWARE_REVISION_MAX_LENGTH];
+  u16 hw_revision = 0;
+  if (file_read_string(DEVICE_HARDWARE_REVISION_FILE_PATH, raw_hw_rev_string,
+                       sizeof(raw_hw_rev_string)) == 0) {
+    hw_revision = (u16)strtod(raw_hw_rev_string, NULL);
+  } else {
+    piksi_log(LOG_ERR, "Error reading hardware revision string (buffer size: %d)", sizeof(raw_hw_rev_string));
+    return -1;
+  }
+
+  switch (hw_revision) {
+    case IMAGE_HARDWARE_UNKNOWN:
+      s = "Unknown";
+      break;
+    case IMAGE_HARDWARE_V3_MICROZED:
+      s = "Piksi Multi MicroZed";
+      break;
+    case IMAGE_HARDWARE_V3_EVT1:
+      s = "Piksi Multi EVT1";
+      break;
+    case IMAGE_HARDWARE_V3_EVT2:
+      s = "Piksi Multi EVT2";
+      break;
+    case IMAGE_HARDWARE_V3_PROD:
+      s = "Piksi Multi";
+      break;
+    default:
+      s = "Invalid";
+      break;
+  }
+
+  if (strlen(s) >= size) {
+    piksi_log(LOG_ERR, "Hardware revision string too large for buffer (size of intended string: %d)", strlen(s));
+    return -1;
+  }
+  strcpy(hw_revision_string, s);
+  return 0;
+}
+
+int hw_variant_string_get(char *hw_variant_string, size_t size)
+{
+  const char *s = NULL;
+
+  if (device_is_duro()) {
+    s = "Duro";
+  } else {
+    s = "Multi";
+  }
+
+  if (strlen(s) >= size) {
+    piksi_log(LOG_ERR, "Hardware variant string too large for buffer (size of intended string: %d)", strlen(s));
+    return -1;
+  }
+  strcpy(hw_variant_string, s);
+  return 0;
+}
+
+int product_id_string_get(char *product_id_string, size_t size)
+{
+  const char *s = NULL;
+
+  if (device_is_duro()) {
+    if (device_has_ins()) {
+      s = "Duro Inertial";
+    } else {
+      s = "Duro";
+    }
+  } else {
+    s = "Multi";
```

Comment:
Add INS check here to allow Piksi Multi Inertial

Reply:
Possible values will be Duro, Duro Inertial, Piksi Multi, Piksi Multi Inertial

Reply:
should match that behavior now

Reply:
Product id was the customer facing piece from what I remember, the hw_* bits were for internal usage like scripts or other daemons.

Reply:
Also thought we could potentially hide/remove the hw_* settings from the console

Reply:
@silverjam correct, this PR covers hiding those https://github.com/swift-nav/piksi_tools/pull/869